sql/moz-fx-data-shared-prod/org_mozilla_ios_firefox/unified_metrics/query.py

```diff
@@ -0,0 +1,143 @@
+"""Generate a query for unifying legacy metrics and glean metrics.
+
+This approach is required because the order of columns is significant in order
+to be able to union two tables, even if the set of columns are the same."""
+
+import requests
+import json
+from bigquery_etl.format_sql.formatter import reformat
+from google.cloud import bigquery, exceptions
+import io
+
+
+def get_columns(schema):
+    """Return a list of columns corresponding to the schema.
+
+    Modified from https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/c2dae92a7ed73e4774a897bc4d2a6a8608875339/mozilla_pipeline_schemas/utils.py#L30-L43
+    """
+
+    def traverse(prefix, columns):
+        res = []
+        for node in columns:
+            name = node["name"]
+            dtype = node["type"]
+            # we consider a repeated field a leaf node, and sorted for our purposes
+            if dtype == "RECORD" and node["mode"] != "REPEATED":
+                res += traverse(f"{prefix}.{name}", node["fields"])
+            else:
+                res += [f"{prefix}.{name} {dtype}"]
+        return res
+
+    res = traverse("root", schema)
+    return sorted(res)
+
+
+def generate_query(columns, table):
```

Comment:
I'm very excited by this function, because it could be a jumping off point to unioning stable data from across all channels of an application (wherein those views don't necessarily have all the same columns, and they are not in the same order).